### PR TITLE
As a User, I should be able to view the details of a Claim

### PR DESCRIPTION
**Pull request by gms-gs**
## Context

Update the claims details page to match the prototype 

prototype: https://itt-mentoring-beta-94341ca35a2c.herokuapp.com/organisations/049fc44e-6876-4884-8a77-d363173ce8d2/claims/34ce9b59-bba4-4714-8f51-d2a93d533e03

## Changes proposed in this pull request

Update the 

## Guidance to review

- Sign in as a normal user
- Create a claim
- Submit the claim
- View the claims index page
- Select the claim

## Link to Trello card

https://trello.com/c/FDAv0ZRT/197-as-a-user-i-should-be-able-to-view-the-details-of-a-claim

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated or [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

## Screenshots

Screenshot 2024-03-05 at 14 25 26

### CODE REVIEW

File: app/views/claims/schools/claims/show.html.erb [hunk 1]
```diff
@@ -5,29 +5,60 @@
...
-      href: edit_claims_school_claim_path(@school, @claim, "mentors"),
-      status: govuk_tag(
-        text: @claim.item_status_tag("mentors").fetch(:text),
-        colour: @claim.item_status_tag("mentors").fetch(:colour),
-      ),
-    ) %>
-  <% end %>
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <h1 class="govuk-heading-l"><%= t(".heading") %> <%= @claim.reference %></h1>
```
Comment (CatalinVoineag) [inline app/views/claims/schools/claims/show.html.erb:10]: Should we rename this to `page_title` and pass the reference to the translation? This way we can use it for the page title as well, which we should add as the page doesn't have a page title.


```suggestion
      <h1 class="govuk-heading-l"><%= t(".page_title" reference: @claim.reference) %></h1>
```

File: app/views/claims/schools/claims/show.html.erb [hunk 2]
```diff
@@ -5,29 +5,60 @@
...
+
+      <%= govuk_summary_list do |summary_list| %>
+        <% summary_list.with_row do |row| %>
+          <% row.with_key(text: t(".school")) %>
```
Comment (CatalinVoineag) [inline app/views/claims/schools/claims/show.html.erb:14]: Should we use `Claim.human_attribute_name` in this file?

```suggestion
          <% row.with_key(text: Claim.human_attribute_name(:school)) %>
```

**Review by CatalinVoineag — COMMENTED** [review]
Minor things

**Review by CatalinVoineag — APPROVED** [review]
Looking good 🚀

File: spec/system/claims/schools/claims/view_claim_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  let!(:claims_mentor) { create(:claims_mentor, first_name: "Barry", last_name: "Garlow") }
+
+  let!(:claim) { create(:claim, school:, reference: "12345678", submitted_at: Time.new(2024, 3, 5, 12, 31, 52, "+00:00"), provider:) }
+  let!(:mentor_training) { create(:mentor_training, claim:, mentor: claims_mentor, hours_completed: 6) }
+
+  scenario "Anne visits the claims index page with a submited" do
+    user_exists_in_dfe_sign_in(user: anne)
+    given_i_sign_in
+    vist_claims_index_page
+    visit_claim_show_page
```
Comment (JamieCleare2525) [inline spec/system/claims/schools/claims/view_claim_spec.rb:24]:
```suggestion
    given_i_visit_the_claim_show_page
```

File: spec/system/claims/schools/claims/view_claim_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    i_can_then_see_the_claim_details
```
Comment (JamieCleare2525) [inline spec/system/claims/schools/claims/view_claim_spec.rb:25]:
```suggestion
    then_i_can_then_see_the_claim_details
```

File: app/views/claims/schools/claims/show.html.erb [hunk 5]
```diff
@@ -1,33 +1,65 @@
...
+              <% @claim.mentors.each do |mentor| %>
+                <li><%= mentor.full_name %></li>
+              <% end %>
+            </ul>
+          <% end %>
+        <% end %>
+
+        <% summary_list.with_row do |row| %>
+          <% row.with_key(text: Claim.human_attribute_name(:date_submitted)) %>
+          <% row.with_value(text: @claim.submitted_at.strftime("%d %B %Y")) %>
```
Comment (JamieCleare2525) [inline app/views/claims/schools/claims/show.html.erb:45]: Not a blocker, but might be worth looking at `en.yml` for date formats

**Review by JamieCleare2525 — APPROVED** [review]

**Pull request merged**